model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/LazyLoader.java

```diff
@@ -33,8 +35,17 @@ public interface LazyLoader<S, D> {
      * Returns data from the given {@code source}. Data is only fetched from {@code source} once and only if necessary, it is
      * up to implementations to decide the momentum to actually fetch data from source.
      *
+     * @param session the session
      * @param source the source from where data will be fetched.
      * @return the data from source
      */
-    D get(Supplier<S> source);
+    D get(KeycloakSession session, Supplier<S> source);
+
+    /**
+     * @deprecated Use {@link #get(KeycloakSession, Supplier)} instead in order to avoid caching partial results when FGAP is enabled
```

Comment:
Looking at the comment "avoid caching partial results when FGAP is enabled" if feels more like this method should be remove altogether, or that it should throw an exception if being called from a context where FGAP filtering would take place. 

Reading the comment, it is not safe to use in such a context.